### DISCUSSION TITLE
dagster dev loads and then just shuts down again

### DISCUSSION THREAD

**Discussion by kienerj** [dagster-io/dagster#14787] [Q&A]
I'm creating my first dagster project. I now wanted to try it out local but when running dagster dev, the service after loading just shuts down again and no dagit is available.

```
(dagster) C:\path\to\my_project>dagster dev -m my_project
2023-06-14 12:55:44 +0200 - dagster - INFO - Loaded environment variables from .env file: API_KEY,TENANT
2023-06-14 12:55:44 +0200 - dagster - INFO - Using temporary directory C:\path\to\my_project\tmpgi0hfgxf for storage. This will be removed when dagster dev exits.
2023-06-14 12:55:44 +0200 - dagster - INFO - To persist information across sessions, set the environment variable DAGSTER_HOME to a directory to use.
2023-06-14 12:55:46 +0200 - dagster - INFO - Launching Dagster services...
2023-06-14 12:55:50 +0200 - dagit - INFO - Loaded environment variables from .env file: API_KEY,TENANT
2023-06-14 12:55:56 +0200 - dagster - INFO - Shutting down Dagster services...
2023-06-14 12:55:56 +0200 - dagster - INFO - Dagster services shut down.
```

This is on windows 10 running in a conda env.

my `Definitions` is in the top level `__init__.py`:

```python
from dagster import Definitions, EnvVar, ScheduleDefinition, define_asset_job
from .assets import entities, last_run
from .resources.entity_search import EntitySearch
from dotenv import load_dotenv

# not sure if this is needed at all given that .env file seems to be loaded twice
load_dotenv()

all_assets = [entities.entities, last_run.last_run]

defs = Definitions(
    assets=all_assets,
    resources={
        "entity_search": EntitySearch(tenant=EnvVar("TENANT"), api_key=EnvVar("API_KEY"))
    },
    schedules=[
        # update assets once per hour
        ScheduleDefinition(
            job=define_asset_job("all_assets", selection="*"), cron_schedule="@hourly"
        ),
    ],
)
```

is this enough? especially the " job=define_asset_job"? Or do I need to explicitly define a job? 
mostly taken form examples throughout the documentation but it's never entirely clear what is really needed to run a full project.

**Comment by OwenKephart** [comment]
Hi @kienerj ! This is definitely odd behavior, I'd expect to at least see some error message before the shut down happened. Which dagster version are you on, and what happens if you just try to run `dagit -m my_project` instead of `dagster dev`?

At a high level, the code you've shared looks totally fine (and even if it weren't, I wouldn't expect the daemons to shut down immediately).

**Reply by kienerj** [reply]
version is 1.3.9 from conda-forge

I get very similar behavior with dagit. it seems to load, and shows a port number:

    2023-06-15 07:26:34 +0200 - dagster.code_server - INFO - Started Dagster code server for module my_project on port 57240 in process 24484

but then it just exits back to cli prompt and there is nothing running on the mentioned port. it also creates the tmxxxxx folders containing the sqllite dbs.

**Comment by kienerj** [comment]
This seem to be an issue with grpc because i found below entries in the Event viewer at the same times I was running dagster dev:

> Faulting application name: python.exe, version: 3.10.8150.1013, time stamp: 0x637c85ed
> Faulting module name: cygrpc.cp310-win_amd64.pyd, version: 0.0.0.0, time stamp: 0x6288e9f8
> Exception code: 0xc0000005
> Fault offset: 0x000000000019ceef
> Faulting process id: 0x3794
> Faulting application start time: 0x01d99f5173242765
> Faulting application path: C:\path\to\anaconda3\envs\dagster\python.exe
> Faulting module path: C:\path\to\anaconda3\envs\dagster\lib\site-packages\grpc\_cython\cygrpc.cp310-win_amd64.pyd
> Report Id: 409b4a84-b028-4ffc-89e2-e5ed31732bb8
> Faulting package full name: 
> Faulting package-relative application ID:

**Reply by OwenKephart** [reply]
Hm good to know -- this seems possible that it might be outside the scope of Dagster specifically.  If you just create an empty definitions file and don't import anything, i.e.

```python
from dagster import Definitions

defs = Definitions()
```

then do you still get this issue?

**Reply by kienerj** [reply]
Still same issue and entry in event viewer. The entire command prompt also becomes partially unresponsive. i can still use it but the close button (x) dosn't react at all and the window needs to be killed with the task manager (or by typing exit)
